common/config/webpack/sampleapp.webpack.config.js

```diff
@@ -5,94 +5,110 @@ const path = require('path');
+        '@internal/chat-component-bindings': path.resolve(sampleAppDir, '../../packages/chat-component-bindings/src'),
+        '@internal/calling-stateful-client': path.resolve(sampleAppDir, '../../packages/calling-stateful-client/src'),
+        '@internal/calling-component-bindings': path.resolve(sampleAppDir, '../../packages/calling-component-bindings/src'),
+        '@internal/acs-ui-common': path.resolve(sampleAppDir, '../../packages/acs-ui-common/src')
+      }
+    },
+    output: {
+      path: path.join(sampleAppDir, env.production ? '/dist/build' : 'dist'),
+      filename: 'build.js'
+    },
+    module: {
+      rules: [
+        {
+          test: /\.tsx?$/,
+          loader: 'ts-loader',
+          options: {
+            transpileOnly: true
+          },
+          exclude: /dist/
+        },
+        {
+          test: /\.css$/i,
+          use: ['style-loader', 'css-loader']
+        },
+        {
+          test: /\.svg/,
+          type: 'asset/inline'
+        }
+      ]
+    },
+    plugins: [
+      new HtmlWebpackPlugin({ template: './public/index.html' }),
+      new webpack.DefinePlugin({
+        'process.env.PRODUCTION': env.production || !env.development,
+        'process.env.NAME': JSON.stringify(require(path.resolve(sampleAppDir, 'package.json')).name),
+        'process.env.VERSION': JSON.stringify(require(path.resolve(sampleAppDir, 'package.json')).version),
+        __CALLINGVERSION__: JSON.stringify(require(path.resolve(sampleAppDir, 'package.json')).dependencies['@azure/communication-calling']),
+        __CHATVERSION__: JSON.stringify(require(path.resolve(sampleAppDir, 'package.json')).dependencies['@azure/communication-chat']),
+        __BUILDTIME__: JSON.stringify(new Date().toLocaleString())
+      }),
+    ],
+    devServer: {
+      port: 3000,
+      hot: true,
+      open: true,
+      contentBase: path.resolve(sampleAppDir, 'public'),
+      proxy: [
+        {
+          path: '/token',
+          target: 'http://[::1]:8080'
+        },
+        {
+          path: '/refreshToken/*',
+          target: 'http://[::1]:8080'
+        },
+        {
+          path: '/isValidThread/*',
+          target: 'http://[::1]:8080'
+        },
+        {
+          path: '/createThread',
+          target: 'http://[::1]:8080'
+        },
+        {
+          path: '/userConfig/*',
+          target: 'http://[::1]:8080'
+        },
+        {
+          path: '/getEndpointUrl',
+          target: 'http://[::1]:8080'
+        },
+        {
+          path: '/addUser/*',
+          target: 'http://[::1]:8080'
+        }
+      ]
     }
-  },
-  output: {
-    path: path.join(sampleAppDir, env.production ? '/dist/build' : 'dist'),
-    filename: 'build.js'
-  },
-  module: {
-    rules: [
-      {
-        test: /\.tsx?$/,
-        loader: 'ts-loader',
+  }
+
+  process.env['FLAVOR'] === 'stable' &&
+    config.module.rules.push({
+      test: /\.tsx?$/,
```

Comment:
does this overwrite the existing rule applied to .tsx? files and say use babel-loader instead of ts-loader, or add a second rule and both are run?

If its the former, then for readability this check would be better inline:
```
const tsloaderRule = {
  test: /\.tsx?$/,
  loader: 'ts-loader',
  options: {
    transpileOnly: true
};

const babelLoaderRule = {
  ...babel load rule config...
};

   ...
    module: {
      rules: [
        process.env['FLAVOR'] === 'stable' ? babelLoaderRule : tsloaderRule,
```

Reply:
This is adding one more rule and dealing with only conditional compile here

Reply:
So will both ts-loader and babel-loader be run on tsx files?

Reply:
Yep, babel-loader will strip conditional @beta code, then it follows our normal build process to get webpack bundles

Reply:
Also for execution order, I guess we might have questions on it:
https://stackoverflow.com/questions/32234329/what-is-the-loader-order-for-webpack#:~:text=%7B%0A%20%20%20%20test%3A%20/%5C.css%24/%2C%0A%20%20%20%20loaders%3A%20%5B%27style%27%5D%2C%0A%7D%2C

It's following a reversed order, so the babel loader will be executed first